packages/type-storage/src/fromMetadata/getHasher.ts

```diff
@@ -2,19 +2,39 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+import { Text } from '@polkadot/types';
 import { StorageHasher } from '@polkadot/types/primitive';
 import { blake2AsU8a, xxhashAsU8a } from '@polkadot/util-crypto';
 
 type HasherInput = string | Buffer | Uint8Array;
 
-type HasherFunction = (data: HasherInput) => Uint8Array;
+export type HasherFunction = (data: HasherInput) => Uint8Array;
+
+function toStorageHasher (text: Text): StorageHasher {
+  switch (text.toString()) {
+    case 'blake2_128':
+      return new StorageHasher('Blake2_128');
+    case 'blake2_256':
+      return new StorageHasher('Blake2_256');
+    case 'twox_128':
+      return new StorageHasher('Twox128');
+    case 'twox_256':
+      return new StorageHasher('Twox256');
+    default:
+      return new StorageHasher(text.toString());
```

Comment:
I think we can throw here instead, not sure what `new StorageHasher(random stuff)` will do

Reply:
yep.